sql/type.go

```diff
@@ -1173,3 +1209,16 @@ func convertArrayForJSON(t arrayT, v interface{}) (interface{}, error) {
 		return nil, ErrNotArray.New(v)
 	}
 }
+
+func hasNullInputs(a interface{}, b interface{}) (bool, int) {
```

Comment:
Would call this compareNulls and document the return values and behavior with a method comment.

Reply:
Great idea!